src/controllers/router/index.tsx

```diff
@@ -1,77 +1,89 @@
-import React, { Component } from 'react';
+import { createMemoryHistory } from 'history';
+import React, { useMemo, useEffect } from 'react';
 
-import { DEFAULT_HISTORY } from '../../common/constants';
-import { getRouterState, RouterContainer } from '../router-store';
-import { UnlistenHistory } from '../router-store/types';
+import { getResourceStore, ResourceContainer } from '../resource-store';
+import {
+  getRouterState,
+  RouterContainer,
+  getRouterStore,
+} from '../router-store';
 
-import { RouterProps } from './types';
+import {
+  RouterProps,
+  MemoryRouterProps,
+  RequestResourcesParams,
+} from './types';
 
-/**
- * Default prop provider for the RouterContainer.
- *
- */
-export class Router extends Component<RouterProps> {
-  static defaultProps = {
-    isStatic: false,
-    isGlobal: true,
-    history: DEFAULT_HISTORY,
-  };
-
-  /**
-   * Keep a copy of the history listener so that we can be sure that
-   * on unmount we call the listener that was in the router store at the
-   * time of mounting.
-   *
-   * This prevents an issue where the wrong listener is removed if the router
-   * is re-mounted.
-   */
-  unlistenHistory: UnlistenHistory | null = null;
-
-  componentDidMount() {
-    if (!this.props.isStatic) {
-      const state = getRouterState();
-      this.unlistenHistory = state.unlisten;
-    }
-  }
-
-  /**
-   * Ensures that the router store stops listening to history when the Router
-   * is unmounted.
-   */
-  componentWillUnmount() {
-    if (this.unlistenHistory) {
-      this.unlistenHistory();
-    }
-  }
+export const Router = ({
+  basePath,
+  children,
+  history,
+  initialRoute,
+  isGlobal = true,
+  onPrefetch,
+  resourceContext,
+  resourceData,
+  routes,
+}: RouterProps) => {
+  useEffect(() => {
+    const { unlisten } = getRouterState();
 
-  render() {
-    const {
-      children,
-      routes,
-      history,
-      initialRoute,
-      isStatic,
-      isGlobal,
-      basePath,
-      resourceContext,
-      resourceData,
-      onPrefetch,
-    } = this.props;
+    return () => {
+      unlisten && unlisten();
+    };
+  }, []);
 
-    return (
+  return (
+    <ResourceContainer isGlobal={isGlobal}>
```

Comment:
Based on our earlier discussion, wonder if `isGlobal` here is really doing what we think. As if `false`, would `getResourceStore()` actually work??

Reply:
Well the integration tests do fail when `isGlobal` is false on the `ResourceContainer` and that is using the public API (i.e. nothing is mocked). Shall I hardcode the resource container `isGlobal` to true?